### ISSUE TITLE
Missing `end` event, ltx parser.

### ISSUE THREAD

**Issue by misterdai** [node-xmpp/ltx#108]
Disclaimer: I may have just missed something.

When using the `ltx/lib/parsers/ltx` parser and processing a stream, there doesn't appear to be an `end` event (or similar) that I can listen for to confirm that parsing has completed.  Any chance of one being added or is it for us to derive?

Cheers

**Comment by sonnyp** [comment]
the `tree` event is available on `https://github.com/node-xmpp/ltx/blob/master/lib/Parser.js`

**Comment by misterdai** [comment]
Ah, does this get fired when streaming data through via something like `fs.createReadStream('./test.xml')`?  Memory usage is a concern for what I'm trying to do so I wanted to avoid having the entire XML data in memory, as I'm processing a large amount of them as quickly as possible.  Hence using the stream approach, which makes me think that if the `tree` event is fired, that data is being held onto until the full data has been parsed.

**Comment by sonnyp** [comment]
> which makes me think that if the tree event is fired, that data is being held onto until the full data has been parsed.

Indeed; `https://github.com/node-xmpp/ltx/blob/master/lib/Parser.js` parses the all tree so data is held into memory.

I'm not against having an `end` event on ltx Parser; feel free to PR.
